### ISSUE TITLE
DITA-OT Homebrew formula for macOS

### ISSUE THREAD

**Issue by infotexture** [dita-ot/dita-ot#3059]
✨ Sean Molenaar (@SMillerDev) has successfully submitted a `dita-ot` formula to [Homebrew](https://brew.sh) with https://github.com/Homebrew/homebrew-core/pull/28762, which allows users to install DITA-OT on macOS via this popular package manager.

The brew formula excludes the Windows batch files (which are useless on macOS) and the legacy `startcmd` scripts, but is otherwise identical to the original distribution package. 

When the package is installed via `brew install dita-ot`, Homebrew automatically downloads the latest version of the toolkit, installs it in a subfolder of the local package Cellar and symlinks the `dita` command to `/usr/local/bin/dita`.

To ensure the formula is kept up-to-date with each DITA-OT release, the core team of @dita-ot/developers should consider updating the formula as part of the release process.

Opening this issue to capture discussion and suggestions, including:

- how to best maintain this formula
- any changes we'd like to make to ensure the formula works like the default distribution package
- how to best promote the new installation option

**Comment by infotexture** [comment]
### Potential changes to the formula

The formula includes an install step that removes the spurious `temp` directory that was present in the original 3.0.4 version of the toolkit on which the formula was originally based, but this folder was removed from DITA-OT 3.1 per https://github.com/dita-ot/dita-ot/issues/2973.

This line should be removed from the next version of the formula:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/dita-ot.rb#L13

Other potential changes to consider:

1. Should the formula retain `startcmd.sh` in case users rely on it?
2. Should the formula set the `$DITA_HOME` environment variable?

As long as the toolkit is invoked solely via the `dita` command, neither should be necessary, but I’m not sure whether that’s a safe assumption yet.

**Comment by infotexture** [comment]
- [ ] Although other Windows batch files are removed, `config/env.bat` currently remains and should also be excluded in the next version of the formula.

**Comment by infotexture** [comment]
### Formula testing

The formula includes a test that uses the `dita` command to build HTML5 output from `docsrc/site.ditamap` from the Homebrew-installed toolkit and verifies whether the `index.html` result file exists (see https://github.com/Homebrew/homebrew-core/blob/master/Formula/dita-ot.rb#L18-L22.)

This test completes fairly quickly, but we could speed things up further by running a simpler functional test that doesn't require as much processing time as the site build.

**Comment by SMillerDev** [comment]
Setting environment variables is not something homebrew formula are allowed to do. You can write a caveat to tell people about it though.

**Comment by infotexture** [comment]
OK, good to know.

**Comment by infotexture** [comment]
#3058 builds on the changes proposed in #3055 to add details on downloading and installing via Homebrew to the [ReadMe](https://github.com/infotexture/dita-ot/blob/feature/update-readme-homebrew/README.markdown).

Assuming we want to promote this as an “official” installation option, I would also add a dedicated topic to the docs with any Homebrew-related specifics to be aware of.

**Comment by infotexture** [comment]
@SMillerDev Any chance you might attend [DITA-OT Day in Rotterdam Nov. 4](https://www.oxygenxml.com/events/2018/dita-ot_day.html)?

If so, would you be interested in giving a lightning talk on installing DITA-OT via Homebrew? 🤔 

Would be a good forum to announce this new option — _and who better to deliver the news?_

**Comment by SMillerDev** [comment]
I don't actually work with dita other than installing it, so probably not.

**Comment by infotexture** [comment]
OK, we'll cover it ourselves then.

**Comment by jelovirt** [comment]
> 1. Should the formula retain startcmd.sh in case users rely on it?

IMO no.

> 2. Should the formula set the $DITA_HOME environment variable?

We if you use dita CLI, you don't need to override `DITA_HOME`. So IMO no.

**Comment by infotexture** [comment]
✨ https://github.com/Homebrew/homebrew-core/pull/32664 was merged, so 3.1.3 is now available via Homebrew:

```bash
$ brew update                                                                                                                                        1475ms
Updated 2 taps (homebrew/cask, homebrew/core).
==> Updated Formulae
dita-ot

$ brew info dita-ot
dita-ot: stable 3.1.3

$ brew install dita-ot
==> Downloading https://github.com/dita-ot/dita-ot/releases/download/3.1.3/dita-ot-3.1.3.zip
🍺  /usr/local/Cellar/dita-ot/3.1.3: 2,314 files, 56.7MB, built in 8 seconds
```

Closing this now. Suggestions for further changes to the formula should be tracked in separate issues.
